index.html

```diff
@@ -129,56 +129,478 @@ <h2 data-dft-for="">
-        an <a data-cite="appmanifest#dfn-object">object</a> that contains
-        information related to note-taking. It has the following members:
+        There are multiple ways that the installation process can be triggered:
       </p>
       <ul>
-        <li>[=note_taking/new_note_url=]
+        <li>An end-user can <dfn data-lt="manual installation">manually</dfn>
+        trigger the installation process through the user agent's
+          <abbr title="User Interface">UI</abbr>, directly invoking the steps
+          to <a>present an install prompt</a>.
+        </li>
+        <li>The installation process can occur through a <dfn>site-triggered
+        install prompt</dfn>: the site can programmatically request that the
+        user agent present an install prompt to the user. The user agent MAY
+        restrict the availability of this feature to cases where, for instance,
+        there are sufficient <a data-cite="appmanifest#installability-signals">
+          installability signals</a> to warrant <a data-cite=
+          "appmanifest#dfn-installed">installation</a> of the web application.
         </li>
       </ul>
+      <p>
+        In any case, the user agent MUST NOT <a>present an install prompt</a>
+        if the document is not <a data-cite=
+        "appmanifest#dfn-is-installable">installable</a>.
+      </p>
+      <p>
+        The user agent MAY, at any time (only if the document is <a data-cite=
+        "appmanifest#dfn-is-installable">installable</a>), run the <a>steps to
+        notify that an install prompt is available</a> at any time, giving the
+        site the opportunity to show a <a>site-triggered install prompt</a>
+        without the user needing to interact with the user agent UI.
+      </p>
+      <p>
+        To <dfn data-lt=
+        "presenting an install prompt|presentation of the install prompt">present
+        an install prompt</dfn>:
+      </p>
+      <ol>
+        <li>Show some user-agent-specific UI, asking the user whether to
+        proceed with installing the app. See <a data-cite=
+        "appmanifest#installation-sec">privacy and security considerations</a>
+        for recommendations relating to this UI. The <var>result</var> of this
+        choice is either {{AppBannerPromptOutcome/accepted}} or
+        {{AppBannerPromptOutcome/dismissed}}.
+        </li>
+        <li>Return <var>result</var>, and <a>in parallel</a>:
+          <ol>
+            <li>If <var>result</var> is {{AppBannerPromptOutcome/accepted}},
+            run the <a>steps to install the web application</a>.
+            </li>
+          </ol>
+        </li>
+      </ol>
+      <p>
+        The <dfn>steps to notify that an install prompt is available</dfn> are
+        given by the following algorithm:
+      </p>
+      <ol>
+        <li>Wait until the {{Document}} of the <a>top-level browsing
+        context</a> is <a data-cite="html#completely-loaded">completely
```

Comment:
"Completely loaded" is not a exported term from HTML, which probably means it's not the right thing to link to?